db/src/metadata.rs

```diff
@@ -105,7 +105,38 @@ impl MetadataReport {
 /// Returns a report summarizing the mutations that were applied.
 pub fn update_attribute_map_from_entid_triples<A, R>(attribute_map: &mut AttributeMap, assertions: A, retractions: R) -> Result<MetadataReport>
     where A: IntoIterator<Item=(Entid, Entid, TypedValue)>,
-          R: IntoIterator<Item=(Entid, Entid, TypedValue)> {
+          R: Iterator<Item=(Entid, Entid, TypedValue)> {
+
+    // Process retractions of schema attributes first. It's allowed to retract a schema attribute
+    // if all of the schema-defining schema attributes are being retracted.
+    // A defining set of attributes is :db/ident, :db/valueType, :db/cardinality.
+    let retractions: Vec<(Entid, Entid, TypedValue)> = retractions.collect();
```

Comment:
Let's put this into the code, next to `SCHEMA_SQL_LIST` -- say, `SCHEMA_RETRACTION_SQL_LIST`.

Reply:
Here you're saying `:db/ident`, but you're not consider `:db/ident` below.  Why not?